posts/post.service.js

```diff
@@ -5,192 +5,191 @@ const isTruthy = require('../util/isTruthy');
-	},
-	update() {
-		return catchAsync(async (req, res, next) => {
-			const doc = await Post.findByIdAndUpdate(req.params.id, req.body, {
-				new: true,
-				runValidators: true,
-			});
+      if (req.user.mongouser.isVoted(doc._id)) {
+        await doc
+          .populate({
+            path: popOptions,
+            model: 'resources',
+            populate: {
+              path: 'images',
+              model: 'image',
+              select: 'name url',
+              populate: {
+                path: 'votes',
+                model: 'Votes',
+                select: 'count  updatedAt'
+              }
+            }
+          })
+          .execPopulate();
+      } else {
+        await doc
+          .populate({
+            path: popOptions,
+            model: 'resources',
+            populate: {
+              path: 'images',
+              model: 'image',
+              select: 'name url'
+            }
+          })
+          .execPopulate();
+      }
+      res.status(200).json({
+        status: 'success',
+        data: doc.toJSONFor(req.user.mongouser)
+      });
+    });
+  },
+  update() {
+    return catchAsync(async (req, res, next) => {
+      const doc = await Post.findByIdAndUpdate(req.params.id, req.body, {
+        new: true,
+        runValidators: true
+      });
 
-			if (!doc) {
-				return next(
-					new AppError('No document found with that ID', 404)
-				);
-			}
-			res.status(200).json({
-				status: 'success',
-				data: {
-					data: doc.toJSONFor(req.user.mongouser),
-				},
-			});
-		});
-	},
-	delete() {
-		return catchAsync(async (req, res, next) => {
-			const doc = await Post.deleteOne({ _id: req.params.id });
-			if (!doc.deletedCount)
-				return next(new AppError('cannot find doc with that id', 404));
-			res.status(204).send();
-		});
-	},
-	getAll(options) {
-		return catchAsync(async (req, res, next) => {
-			let data = Post.find();
-			if (options.getRecentFirst) {
-				data.sort('-createdAt');
-			}
-			if (options.populateResources) {
-				data.populate({
-					path: 'resources',
-					model: 'resources',
-					populate: {
-						path: 'images',
-						model: 'image',
-						select: 'name url',
-					},
-				});
-			}
-			if (options.populateAuthor) {
-				data.populate('author', 'name email');
-			}
-			data = await data;
+      if (!doc) {
+        return next(new AppError('No document found with that ID', 404));
+      }
+      res.status(200).json({
+        status: 'success',
+        data: {
+          data: doc.toJSONFor(req.user.mongouser)
+        }
+      });
+    });
+  },
+  delete() {
+    return catchAsync(async (req, res, next) => {
+      const query = await Post.findById(req.params.id);
+      if (query) {
+        if (String(query.get('author')) === String(req.user.mongouser._id)) {
+          await Post.deleteOne({ _id: req.params.id });
+          return res.status(204).send();
+        }
+        return next(new AppError("Only post's owner can delete it", 403));
```

Comment:
same as [this comment](https://github.com/m3ntorship/pickly-posts/pull/35/files?diff=split&w=1#r515560003)